fdbclient/ClientKnobs.h

```diff
@@ -120,6 +120,14 @@ class ClientKnobs : public KnobsImpl<ClientKnobs> {
 	int64_t CORE_VERSIONSPERSECOND; // This is defined within the server but used for knobs based on server value
 	int LOG_RANGE_BLOCK_SIZE;
 	int MUTATION_BLOCK_SIZE;
+	double MAX_VERSION_CACHE_LAG; // The upper bound in seconds for OK amount of staleness when using a cached RV
```

Comment:
In the Record Layer GRV cache implementation, we made this option something that could be set on a per transaction basis. That is, the user provided a `WeakReadSemantics` object to the transaction that specified how stale the transaction read version was allowed to be (see: https://github.com/FoundationDB/fdb-record-layer/blob/5efeb2355be9507a9802d18439e53016dfc1cac3/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java#L1314), and then the read version assigner could look at the staleness to determine whether the cached version was appropriate to use or not (see: https://github.com/FoundationDB/fdb-record-layer/blob/5efeb2355be9507a9802d18439e53016dfc1cac3/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java#L436).

I'm not sure I could give you a totally coherent example of why some users might want to set different staleness bounds for different transactions, but it does seem like something the user might want to control in general, and I think knobs are a little brittle for that. Ideally, this would be an option the client could set, I think.